File: website/docs/tutorials/authentication-and-security/cors.py
# skip:start
from typegraph_next import typegraph, Policy, t, Graph
from typegraph_next.graph.params import Cors
from typegraph_next.runtimes.random import RandomRuntime


# skip:end
@typegraph(
    # highlight-next-line
    cors=Cors(
        # highlight-next-line
        allow_origin=["https://not-this.domain"],
        # highlight-next-line
        allow_headers=["x-custom-header"],
        # highlight-next-line
        expose_headers=["header-1"],
        # highlight-next-line
        allow_credentials=True,
        # highlight-next-line
        max_age_sec=60,
        # highlight-next-line
    ),
)
def auth(g: Graph):
    random = RandomRuntime(seed=0)
    public = Policy.public()

    g.expose(
        public,
        catch_me_if_you_can=random.gen(t.string()),
    )

File: website/typegraphs/cors.ts
// skip:start
import { Policy, t, typegraph } from "@typegraph/sdk/index.js";
import { RandomRuntime } from "@typegraph/sdk/runtimes/random.js";

// skip:end

typegraph({
  name: "auth",
  // highlight-next-line
  cors: {
    // highlight-next-line
    allowOrigin: ["https://not-this.domain"],
    // highlight-next-line
    allowHeaders: ["x-custom-header"],
    // highlight-next-line
    exposeHeaders: ["header-1"],
    // highlight-next-line
    allowCredentials: true,
    // highlight-next-line
    maxAgeSec: 60,
    // highlight-next-line
    allowMethods: ["GET"],
  },
}, (g) => {
  const random = new RandomRuntime({ seed: 0 });
  const pub = Policy.public();

  g.expose({
    catch_me_if_you_can: random.gen(t.string()).withPolicy(pub),
  });
});
